src/Engine/GameAction.java

```diff
@@ -93,6 +118,8 @@ public boolean isReadyToExecute()
    * Performs the built-up action, using the passed-in GameMap.
    * @return true if the action successfully executes, false if a problem occurs.
    */
+  // TODO: Should this validate that the GameAction is valid (e.g. we aren't trying
+  //  to move an Infantry unit 10 spaces)?
```

Comment:
I think isReadyToExecute should do it, if anything does. Do we really need more tests for valid action? I think the systems we have in place already should probably be enough.